lib/api/utilities/functions.php

```diff
@@ -81,172 +81,173 @@ function beans_remove_dir( $dir_path ) {
+		return isset( $original_url ) ? $original_url : $url;
 	}
 
 	// Parse url and standardize backslashes.
-	$url = parse_url( $url, PHP_URL_PATH );
+	$url  = parse_url( $url, PHP_URL_PATH );
 	$path = wp_normalize_path( $url );
-	$explode = beans_get( 0, explode( '/' , trailingslashit( ltrim( $path, '/' ) ) ) );
 
-	// Maybe remove tilde from path.
-	if ( false !== stripos( $explode, '~' ) ) {
-		$path = preg_replace( '#\~[^/]*\/#', '', $path );
+	// Maybe remove tilde from path. -> Issue #66 - Fixed.
+	$trimmed_path = trailingslashit( ltrim( $path, '/' ) );
+	$maybe_tilde  = beans_get( 0, explode( '/', $trimmed_path ) );
+	if ( false !== stripos( $maybe_tilde, '~' ) ) {
+		$ends_with_slash = substr( $path, - 1 ) === '/';
+		$path            = preg_replace( '#\~[^/]*\/#', '', $trimmed_path );
+		if ( $path && ! $ends_with_slash ) {
+			$path = rtrim( $path, '/' );
+		}
 	}
 
 	// Set root if it isn't cached yet.
-	if ( ! $root ) {
-
+	if ( ! $root_path ) {
 		// Standardize backslashes and remove windows drive for local installs.
-		$root = wp_normalize_path( untrailingslashit( ABSPATH ) );
-		$set_root = true;
-
+		$root_path = wp_normalize_path( untrailingslashit( ABSPATH ) );
+		$set_root  = true;
 	}
 
-	// Remove subfolder if necessary.
-	if ( '' !== ( $subfolder = parse_url( site_url(), PHP_URL_PATH ) ) ) {
-
-		// Set root if it isn't cached.
-		if ( isset( $set_root ) ) {
-
-			// Remove subfolder.
-			$root = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root );
+	// If the subfolder exists for the root URL, then strip it off of the root path.
+	// Why? We don't want a double subfolder in the final path.
+	$subfolder = parse_url( $site_url, PHP_URL_PATH );
+	if ( isset( $set_root ) && $subfolder && '/' !== $subfolder ) {
+		$root_path = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root_path );
 
-			// Add an extra step which is only used for extremely rare case.
-			if ( defined( 'WP_SITEURL' ) && '' !== ( $subfolder = parse_url( WP_SITEURL, PHP_URL_PATH ) ) ) {
-				$root = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root );
+		// Add an extra step which is only used for extremely rare case.
+		if ( defined( 'WP_SITEURL' ) ) {
+			$subfolder = parse_url( WP_SITEURL, PHP_URL_PATH );
+			if ( '' !== $subfolder ) {
+				$root_path = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root_path );
 			}
 		}
+	}
 
-		// Remove the blog path for multsites.
-		if ( ! is_main_site() ) {
-
-			// Set blogdetails if it isn't cached.
-			if ( ! $blogdetails ) {
-				$blogdetails = get_blog_details( get_current_blog_id() );
-			}
-
-			$path = preg_replace( '#^(\/?)' . trailingslashit( preg_quote( ltrim( $blogdetails->path, '/' ) ) ) . '#', '', $path );
+	// Remove the blog path for multisites.
+	if ( ! is_main_site() ) {
```

Comment:
Just like in the other function, I moved the multi-site task out of the subfolder task. See note in `beans_path_to_url`